pkg/ottl/functions.go

```diff
@@ -93,105 +96,87 @@ func (p *Parser[K]) buildArgs(inv invocation, fType reflect.Type) ([]reflect.Val
 	return args, nil
 }
 
-func (p *Parser[K]) buildSliceArg(inv invocation, argType reflect.Type, startingIndex int, args *[]reflect.Value) error {
+func (p *Parser[K]) buildSliceArg(inv invocation, argType reflect.Type, index int, args *[]reflect.Value) error {
```

Comment:
I like how you changed `buildArg` to return the value and handle the appending in `buildArgs`.  Can this function do the same?

Reply:
Good idea. I've updated the logic so that assembling the args list is now done in `buildArgs`.